contributed_definitions/nyaml/NXraman.yaml

```
@@ -0,0 +1,755 @@
+        device:
+          doc: |
+            This should contain the name for each halfwave plate used in the
+            NXwaveplate group.
+      waveplate_NAME(NXwaveplate):
+        doc: |
+          Properties of the device for the manipulation of the state of light which is a half wave
+          plate.
+        type:
+          enumeration: [objective, lens, glass fiber, none, other]
+      polarization_filter(NXbeam_device):
+        filter_mechanism(NX_CHAR):
+          exists: optional
+          doc: |
+            Physical principle of the polarization filter used to create a
+            defined incident or scattered light state.
+          enumeration: [Polarization by Fresnel reflection, Birefringent polarizers, Thin film polarizers, Wire-grid polarizers, other]
+        specific_polarization_filter_type(NX_CHAR):
+          doc: |
+            Specific name or type of the polarizer used.
+            
+            Free text, for example: Glan-Thompson, Glan-Taylor, Rochon Prism, Wollaston
+            Polarizer...
+      spectral_filter_N(NXbeam_device):
+        filter_type:
+          exists: optional
+          doc: |
+            Type of laserline filter used to supress the laser, if measurements
+            close to the laserline are performed.
+          enumeration: [long-pass filter, short-pass filter, Notch filter, reflection filter, neutral density filter, other]
+        intended_use:
+          exists: optional
+          doc: |
+            Type of laserline filter used to supress the laser, if measurements
+            close to the laserline are performed.
+          enumeration: [laser line cleanup, raylight line removal, spectral filtering, intensity manipulation, other]
+        filter_characteristics(NX_NUMBER):
+          exists: optional
+          doc: |
+            Properties of the spectral filter such as wavelength dependent Transmission
+            or reflectivity.
+          dimensions:
+            rank: 2
+            dim: [[1, 2], [2, N_spectrum_filter]]
+            # 2 -> [Wavelength, Transmission]
+            # N_spectrum_filter --> Number of wavelengths
+          \@characteristics_type:
+            exists: optional
+            doc: |
+              Which property is used to form the spectral properties of light,
+              i.e. transmission or reflection properties.
+            enumeration: [transmission, reflection]
+      opt_monochromator(NXbeam_device):
+        exists: optional
+        doc: |
+          Dispersive element to create e.g. an energetic resolution used in the 
+          measurement.
+        device:
+          doc: |
+            It should contain the name of the Nexus Monochromator group
+      (NXmonochromator):
+        doc: |
+          Monochromator with arbitrary choice of the name, which shall be referenced
+          by the spectrometer(NXbeam_device).
+      Sample_as_beam_element(NXbeam_device):
```

```suggestion
      sample_as_beam_element(NXbeam_device):
```

I think it's a typo or is it meant to also contain names like `bananample_as_beam_element`?

---

yea - this sounds good. Maybe in a future project we will need this :x (yea its a typo)